internal/pkg/config/testdata/src/configv2/discussion/test-config-v2.yaml

```diff
@@ -0,0 +1,66 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source:
+  value:
+    - id: "Tokens are secret"
+      package: configv2/discussion
+      type: Token
+    - id: "TokenBundles are entirely secret"
+      package: configv2/discussion
+      type: TokenBundle
+    - id: "NamedToken.token is secret, NamedToken.name is not"
+      package: configv2/discussion
+      type: NamedToken
+      fieldName: token
+    - id: "Tagged fields are secret, neighboring untagged fields are not"
+      # empty package and type fields match anything
+      fieldTag:
+      - myTag: source
+    - id: "In context, the parameter of SaveSecret(secret interface{}) is a source"
```

Comment:
I'm a little worried that such cases could be very frequent in some code bases and that they would lead to an explosion in the size of the config, which could make it hard for users to manage.

It seems to me that this is something we should handle via IPA (**I**nter**P**rocedural **A**nalysis).

Reply:
Not to discount the value of interprocedural work, but this would cover the case of, e.g., a `[]byte` stream that is expected to contain a secret.  Interprocedural would cover the case that this byte stream were either `type SecretStream []byte` or produced out of scope by some `var secret interface{} = os.Getenv("PASSWORD") ; SaveSecret(secret)`.

I agree that, by itself, it is insufficient to cover our concern about interface obfuscation.

Reply:
I think in short, to me it seems more robust to match calls than to match functions, because a call matcher can potentially be used in many places. It's also easier for users to just say "this function produces a source" than to look for the potentially many places where a value tainted from the result of a source-producing function is used.

In your example, I'm thinking that `[]byte` must have come from somewhere, and in general I think it would be preferable to look for the location where it is produced.

Being able to match a specific parameter of a specific function may be useful if other options don't work, though.